### DISCUSSION TITLE
Noob w/question for Ring Doorbell 2

### DISCUSSION THREAD

**Discussion by KruseLuds** [tsightler/ring-mqtt#832] [Support Q&A] [answered]
I have been horrendously dissatisfied with the regular Ring Doorbell integration for HA. The doorbell does detect motion very quickly, but triggers related to that motion in HA are only tripped maybe half of the time, and when they are triggered, most of those are more than a minute after the doorbell sensed the actual motion. (Indicator LEDs on the Ring Doorbell 2 light up when motion is sensed and those always appear instantly.)  I have no other Ring equipment at this point...

Can this replace the ring doorbell integration for me with less latency? If it is much quicker,, even if it is a bunch of work I'll set it up! 

1. Can I use this in place of or in addition to the HA Ring Doorbell integration?

2. I have disabled the Ring Doorbell Integration and installed this and logged into it with 2fa and saw a lot of activities in the addon log so I should be good to go... What is my next step to switch my integrations over to this?

If anyone can point me in the right direction I would be very thankful - !

**Comment by KruseLuds** [comment]
I can add some context here as I am getting closer to an answer and the source of the issue, anythinig anyone can add here would be of great help. I actually now am using both this and the traditional ring integration, to compare how quickly each report motion. Ring-MQTT almost always reports back into home assistant within 1 or 2 seconds of the actual motion (when it is reported by the doorbell) taking place. The ring integration is consistently 3 seconds behind that, so this shaves 3 secods off of the issue.

The only device that is turned from the motion at night (as well as turned off at certain times, and back on at sunset, yadiyadiyada, is a wemo in wall switch by the front door for outside lights). That is cloud based and takes quite a while to turn on and does not work consistently when called by automations.  My WiFi network is very fast with a strong signal everywhere so WiFi is not the issue. I have had latency issues with other Wemo products and am slowly replacing them to also get away from the cloud wherever I can.

Long story short, I am replacing that in-wall light switch with a Leviton D215S and will be using matter to conmtrol it. I have their dimmer version of this switch in another room and when I got rid of the cloud based Leviton integration and switched to matter for that it has been flawless and rock solid ever since.

I will report back after replacing the wall switch with one matter-based. I have tried alot of things with Ring to try to resolve the latency issue in the past (sometimes it would not even recognize motion) - so if this does not resolve it I will investigate further but may move away from Ring also.... Has anyone done a cpomparison on the motion sensing latency with the older Doorbell 2 model as opposed to the newer Ring doorbells?

Thoughts?

**Reply by tsightler** [reply]
As far as I can tell, the native HA Ring integration still uses polling to collect events (I previously thought this had changed, but, upon further inspection, it appears it has not)  This polling is every 5 seconds , while ring-mqtt uses push notifications, so notification time should be basically the same speed as you receive a notification on your device from the Ring app or browser based web UI.

Ring doorbells don't really use much special technology, the newer cams don't seem that different than old ones as far as initial detection latency, and you indicated that this was already fine.  Once Ring implemented push notifications and we were able to get that into ring-mqtt, most complaints about notification latency went away.  I've been using this setup for 5+ years including 4 cameras in the front yard that detect motion from various angles and then turn on floodlights and a porch light via a Tasmota Wifi switch and latency is sub-second.

However, Ring devices are cloud powered and Ring services no stranger to delayed notifications, although reliability had been much better in the last couple of years than it was in their earlier days.  The cheaper/battery powered Ring devices are also power optimized vs performance, so they have slightly higher latency, but it's not a huge difference.  The use of Firebase Cloud Messaging for push notification is quite standard and generally works very well, but it's still cloud, there will be some delay.

The biggest issue is with smart alerts as it seems most people don't realize that smart alerts are not per-device, so if you set smart notifications to only notify on person detected on your phone it also means that notifications are delivered to ring-mqtt only when a person is detected.  On the other hand the polling in the native Ring HA integration isn't impacted by smart alerts since it's not dependent on push notifications, but it will always be significantly slower.

**Comment by KruseLuds** [comment, marked as answer]
@tsightler, thank you for your insight. I have replaced my front door light switch (Wemo) with the Leviton DS215S, and connected it to HA via Matter, and updated all of the related automations accordingly (which I have also switched over from using the Ring integration to using the Ring-MQTT Add-On). 

One of the automations is that if motion is sensed by the ring doorbell, and the front lights are off and the outside light as detected is less than 5,000 lumens by my Ambient Weather WS-2902 weather station, then this front door light switch will be turned on. The ring doorbell is so close to the edge of the front door that sometimes opening of the door will detect the motion.

In the attached video, it is night and I have turned off the front door light switch - and then open the door which would trigger the aforementioned automation.

I am absolutely flabbergasted by the speed of the setup now and am completely thrilled.

Watch the video - Mic drop - case CLOSED (THANK YOU)!


https://github.com/tsightler/ring-mqtt/assets/96640689/0811f609-eb0b-4c30-b937-ddeca8c068cb

FYI - ALL of my automations relying on the Ring doorbell now work this well, and I have repeated the tests over and over - the speed is always like this, it is not just fast sometimes, it is now always this same speed. Problems gone.
